client/reader/user-profile/components/user-profile-header/test/index.tsx

```diff
@@ -0,0 +1,213 @@
+/**
+ * @jest-environment jsdom
+ */
+
+import { render, screen } from '@testing-library/react';
+import React from 'react';
+import UserProfileHeader from '../index';
+
+// Mock external icon
+jest.mock( '@wordpress/icons', () => ( {
+	external: 'mock-external-icon',
+	Icon: ( { icon } ) => <span data-testid="icon">{ icon }</span>,
+} ) );
+
+// Mock ReaderAvatar component
+jest.mock( 'calypso/blocks/reader-avatar', () => ( {
+	__esModule: true,
+	default: ( { author, iconSize } ) => (
+		<div data-testid="reader-avatar" data-author-id={ author.ID } data-icon-size={ iconSize }>
+			{ author.avatar_URL && (
+				<img src={ author.avatar_URL } alt="avatar" data-testid="avatar-img" />
+			) }
+		</div>
+	),
+} ) );
+
+// Mock SectionNav components
+jest.mock( 'calypso/components/section-nav', () => ( {
+	__esModule: true,
+	default: ( { children } ) => <div data-testid="section-nav">{ children }</div>,
+} ) );
+
+// Mock SectionNav/Tabs component
+jest.mock( 'calypso/components/section-nav/tabs', () => ( {
+	__esModule: true,
+	default: ( { children } ) => <div data-testid="nav-tabs">{ children }</div>,
+} ) );
+
+// Mock SectionNav/Item component
+jest.mock( 'calypso/components/section-nav/item', () => ( {
+	__esModule: true,
+	default: ( { children, path, selected } ) => (
+		<a href={ path } data-testid="nav-item" data-selected={ selected ? 'true' : 'false' }>
+			{ children }
+		</a>
+	),
+} ) );
+
+describe( 'UserProfileHeader', () => {
+	const defaultUser = {
+		ID: 123,
+		user_login: 'testuser',
+		display_name: 'Test User',
+		avatar_URL: 'https://example.com/avatar.jpg',
+		profile_URL: 'https://wordpress.com/testuser',
+		bio: undefined,
+	};
+
+	/**
+	 * Test: Avatar rendering
+	 *
+	 * Verifies that:
+	 * 1. The avatar component is rendered with correct props
+	 * 2. The correct user data is passed to the avatar component
+	 */
+	test( 'should render the avatar with correct user information', () => {
+		render( <UserProfileHeader user={ defaultUser } /> );
+
+		const avatars = screen.getAllByTestId( 'reader-avatar' );
+		// @ts-expect-error -- jest-dom matchers are available globally
+		expect( avatars[ 0 ] ).toBeInTheDocument();
+		// @ts-expect-error -- jest-dom matchers are available globally
+		expect( avatars[ 0 ] ).toHaveAttribute( 'data-author-id', defaultUser.ID.toString() );
+
+		// Test that desktop and mobile versions are properly rendered
+		const desktopAvatar = document.querySelector(
+			'.user-profile-header__avatar-desktop [data-testid="reader-avatar"]'
```

Comment:
I think it will be better if we update `data-testid` here.

Reply:
Sorry, I don't follow. Can you elaborate more?

Reply:
Can we only use data-testid for the selector by screen.getByTestId?

Reply:
Oh, you mean like this: e54db48825f028405618566dd51ff63320401805